src/backend/distributed/commands/call.c

```diff
@@ -0,0 +1,205 @@
+	{
+		ereport(DEBUG1, (errmsg("cannot push down invalid distribution_argument_index")));
+		return false;
+	}
+
+	if (contain_volatile_functions((Node *) funcExpr->args))
+	{
+		ereport(DEBUG1, (errmsg("arguments in a distributed stored procedure must "
+								"be constant expressions")));
+		return false;
+	}
+
+	distTable = DistributedTableCacheEntry(colocatedRelationId);
+	partitionColumn = distTable->partitionColumn;
+	if (partitionColumn == NULL)
+	{
+		/* This can happen if colocated with a reference table. Punt for now. */
+		ereport(DEBUG1, (errmsg(
+							 "cannot push down CALL for reference tables")));
+		return false;
+	}
+
+	partitionValue = (Const *) list_nth(funcExpr->args, procedure->distributionArgIndex);
+	if (!IsA(partitionValue, Const))
+	{
+		ereport(DEBUG1, (errmsg("distribution argument value must be a constant")));
+		return false;
+	}
+
+	partitionValueDatum = partitionValue->constvalue;
+	if (partitionValue->consttype != partitionColumn->vartype)
+	{
+		CopyCoercionData coercionData;
+
+		ConversionPathForTypes(partitionValue->consttype, partitionColumn->vartype,
+							   &coercionData);
+
+		partitionValueDatum = CoerceColumnValue(partitionValueDatum, &coercionData);
+	}
+
+	shardInterval = FindShardInterval(partitionValueDatum, distTable);
+	if (shardInterval == NULL)
+	{
+		ereport(DEBUG1, (errmsg("cannot push down call, failed to find shard interval")));
+		return false;
+	}
+
+	placementList = FinalizedShardPlacementList(shardInterval->shardId);
+	if (list_length(placementList) != 1)
+	{
+		/* punt on this for now */
+		ereport(DEBUG1, (errmsg(
+							 "cannot push down CALL for replicated distributed tables")));
+		return false;
+	}
+
+	placement = (ShardPlacement *) linitial(placementList);
+	workerNode = FindWorkerNode(placement->nodeName, placement->nodePort);
+	if (workerNode == NULL || !workerNode->hasMetadata || !workerNode->metadataSynced)
+	{
+		ereport(DEBUG1, (errmsg("there is no worker node with metadata")));
+		return false;
+	}
+
+	ereport(DEBUG1, (errmsg("pushing down the procedure")));
+
+	/* build remote command with fully qualified names */
+	callCommand = makeStringInfo();
+	appendStringInfo(callCommand, "CALL %s", pg_get_rule_expr((Node *) funcExpr));
+
+	{
+		Tuplestorestate *tupleStore = tuplestore_begin_heap(true, false, work_mem);
+		TupleDesc tupleDesc = CallStmtResultDesc(callStmt);
+		TupleTableSlot *slot = MakeSingleTupleTableSlotCompat(tupleDesc,
+															  &TTSOpsMinimalTuple);
+		Task *task = CitusMakeNode(Task);
+
+		task->jobId = INVALID_JOB_ID;
+		task->taskId = 0;
+		task->taskType = DDL_TASK;
+		task->queryString = callCommand->data;
+		task->replicationModel = REPLICATION_MODEL_INVALID;
+		task->dependedTaskList = NIL;
+		task->anchorShardId = placement->shardId;
+		task->relationShardList = NIL;
+		task->taskPlacementList = placementList;
+
+		ExecuteTaskListExtended(ROW_MODIFY_NONE, list_make1(task),
```

Comment:
super minor: instead of passing `true`, it'd be nice to pass `hasReturning` variable, because I've trying to understand how does this return tuples with `ROW_MODIFY_NONE ` and `DDL_TASK`